packages/bun-transpiler/src/index.ts

```diff
@@ -2,52 +2,48 @@ import Bun from 'bun'
 import { createMiddleware } from 'hono/factory'
 
 type BunTranspilerOptions = {
-  extensions: string[]
-  headers: HeadersInit
-  transpilerOptions: Bun.TranspilerOptions
+  extensions?: string[]
+  headers?: Record<string, string | string[]>
+  transpilerOptions?: Bun.TranspilerOptions
 }
 
-export const bunTranspiler = (
-  options: BunTranspilerOptions = {
-    extensions: ['.ts', '.tsx'],
-    headers: { 'content-type': 'application/javascript' },
-    transpilerOptions: {
-      minifyWhitespace: true,
-      target: 'browser',
-    },
-  }
-) => {
+const DEFAULT_OPTIONS: BunTranspilerOptions = {
```

Comment:
How about exporting `DEFAULT_OPTIONS` and having users use it?

Like this:

```ts
import { defaultOptions } from '@hono/bun-transpiler'

app.use(
  '*',
  bunTranspiler({
    extensions: ['mts', ...defaultOptions.extensions],
  })
)
```

For this reason, I would like to do these.

* Because users use it, so rename it to `defaultOptions`.
* Set the type as `Required<BunTranspilerOptions>`.

Reply:
That's a good point! I like that suggestion. Will expose `deufaltOptions`. I'm not convinced about requiring it as an input tough because why would I force the developer to provide me with the defaultOptions? Or did I understand your second suggestion wrong @yusukebe ?

Reply:
I'm sorry! I didn't explain enough. As it is now, we get the following error.

<img width="505" alt="Screenshot 2024-01-10 at 7 06 58" src="https://github.com/honojs/middleware/assets/10682/d3999075-a168-47d8-87b0-8212815c4b4c">

This is because the property is optional. So you can avoid the error by adding `Required` to the `defaultOptions` type as follows.

```ts
export const defaultOptions: Required<BunTranspilerOptions> = {
  extensions: ['.ts', '.tsx'],
  //...
}
```

Reply:
Oh! Now I understand. Didn't check that. Good catch! Adjusted to your suggested solution.